contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -85,4 +130,80 @@ contract TrueCreditAgency is UpgradeableClaimable {
         pool.token().safeApprove(address(pool), amount);
         pool.repay(amount);
     }
+
+    function poke(ITrueFiPool2 pool) public {
+        for (uint16 i = 1; i <= MAX_CREDIT_SCORE; i++) {
+            CreditScoreBucket storage bucket = buckets[pool][i];
+            if (bucket.borrowersCount == 0) {
+                continue;
```

Comment:
Won't that be super costly?

Reply:
Maybe we somehow store min and max credit score that we currently have in use and restrict iteration to that range?

Reply:
We could also use a sorted array of buckets, with max size 256. It would only contain the scores that are already set, with a worst case that behaves like this.

Reply:
I used a bit set to mark used buckets in #bucket-2 branch and looks like this could improve the gas cost a lot (6 times in a simple case)

Reply:
That's awesome! Yeah LoC is looking very gas-hungry. I think we'll have to do some profiling after all the pieces are connected to see if there are optimizations like this that we've left on the table.